### PR TITLE
Fix#453: Added fsPromises.link() tests

### PR DESCRIPTION
**Pull request by Dragomegak**
Implemented the promise tests of link to the fs.link.spec.js based on the non promised version

### CODE REVIEW

File: tests/spec/fs.link.spec.js [hunk 1]
```diff
@@ -111,3 +111,119 @@ describe('fs.link', function() {
...
+  it('should create a link to an existing file', function(done) {
+    var fs = util.fs();
+
+    fs.open('/myfile', 'w+', function(error, fd) {
+      if(error) throw error;
+
+      fs.close(fd, function(error) {
+        if(error) throw error;
+
+        fs.promises.link('/myfile', '/myotherfile', function(error) {
```
Comment (dnguneratne) [inline tests/spec/fs.link.spec.js]: Should follow the promise syntax
```js
fs.promises.link('/myfile', '/myotherfile')
	.then()..
```

File: tests/spec/fs.link.spec.js [hunk 2]
```diff
@@ -111,3 +111,119 @@ describe('fs.link', function() {
...
+
+
+
+
+describe('fs.promises.link', function() {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should be a function', function() {
+    var fs = util.fs();
```
Comment (dnguneratne) [inline tests/spec/fs.link.spec.js]: You can use the promise API directly with:
`var fs = util.fs().promises;`

File: tests/spec/fs.link.spec.js (hunk 1 above)
Comment (Dragomegak) [reply tests/spec/fs.link.spec.js]: Done for tests:  
- create a link to an existing file
- not follow symbolic links 

although I could not get it running for not allowing links to a directory

File: tests/spec/fs.link.spec.js (hunk 2 above)
Comment (Dragomegak) [reply tests/spec/fs.link.spec.js]: I tried checking out other implementations of this so I could see how I could do this but other people's code did not do this. By chance do you mind giving me more information about how to implement it?

Comment (humphd) [reply tests/spec/fs.link.spec.js]: I think it's fine either way.  The `util.fs().promises` way would be good if you were needing to use it multiple times, and you could cache it like this.  In this case, it's just once, so either or is fine.

File: tests/spec/fs.link.spec.js [hunk 3]
```diff
@@ -111,3 +111,121 @@ describe('fs.link', function() {
...
+  });
+
+  it('should create hard link to identical data node', function(done) {
+    var fs = util.fs();
+    var contents = 'Hello World!';
+
+    fs.writeFile('/file', contents, function(err) {
+      if(err) throw err;
+
+      fs.promises.link('/file', '/hlink', function(err) {
```
Comment (humphd) [inline tests/spec/fs.link.spec.js]: This is mixing callbacks and promises.  You shouldn't be passing a callback to link here.

File: tests/spec/fs.link.spec.js [hunk 4]
```diff
@@ -111,3 +111,88 @@ describe('fs.link', function() {
...
+
+  it('should be a function', function() {
+    var fs = util.fs();
+    expect(fs.promises.link).to.be.a('function');
+  });
+
+  it('should create a link to an existing file', function(done) {
+    var fs = util.fs();
+
+    fs.open('/myfile', 'w+', function(error, fd) {
```
Comment (humphd) [inline tests/spec/fs.link.spec.js]: You're mixing callbacks and promises in the same test, which is a bit confusing.  Here's how you should write something like this:

```js
var fsPromises = util.fs().promises;
var oldStats;

return fsPromises.open('/myfile', 'w+')
  .then(fd => fsPromises.close(fd))
  .then(() => fsPromises.stat('/myfile'))
  .then(result => {
    oldStats = result;
    return fsPromises.stat('/myotherfile')
  })
  .then(result => {
    // do all your expect assertions here)
  });
```

Can you fix all your tests so they only use promises?

**Comment by humphd** [comment]
Fixed, rebased, and landed in https://github.com/filerjs/filer/commit/16e6b3e1c9ba9840dba0815ef3af1c9d459e1897.

**Pull request closed without merging**